lib/xebow_web/live/matrix_live.ex

```diff
@@ -0,0 +1,191 @@
+defmodule XebowWeb.MatrixLive do
+  use XebowWeb, :live_view
+
+  alias RGBMatrix.{Animation, Engine}
+
+  @layout Xebow.layout()
+  @black Chameleon.HSV.new(0, 0, 0)
+  @black_frame @layout
+               |> Layout.leds()
+               |> Map.new(fn led ->
+                 {led.id, @black}
+               end)
+
+  @impl Phoenix.LiveView
+  def mount(_params, _session, socket) do
+    if connected?(socket) do
+      register_with_engine!()
+    end
+
+    {config, config_schema} = Engine.get_animation_config()
+
+    initial_assigns = [
+      leds: make_view_leds(@black_frame),
+      config: config,
+      config_schema: config_schema,
+      animation_types: Animation.types(),
+      current_animation_index: 0
+    ]
+
+    {:ok, assign(socket, initial_assigns)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info({:render, frame}, socket) do
+    {:noreply, assign(socket, leds: make_view_leds(frame))}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info({:render_config, {config, config_schema}}, socket) do
+    {:noreply, assign(socket, config: config, config_schema: config_schema)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("key_pressed", %{"key-id" => id_str}, socket) do
+    key_id = String.to_existing_atom(id_str)
+
+    case Layout.led_for_key(@layout, key_id) do
+      nil -> :noop
+      led -> Engine.interact(led)
+    end
+
+    {:noreply, socket}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_event("update_config", %{"_target" => [field_str]} = params, socket) do
+    value = Map.fetch!(params, field_str)
+
+    Engine.update_animation_config(%{field_str => value})
+
+    {:noreply, socket}
+  end
+
+  # TODO: the following two are duplicated between `Keyboard` and here.
+
+  @impl Phoenix.LiveView
+  def handle_event("next_animation", %{}, socket) do
+    next_index = socket.assigns.current_animation_index + 1
+
+    next_index =
```

Comment:
do you think it reads cleaner to do `next_index = mod(socket.assigns.current_animation_index + 1, len(socket.assigns.animation_types))` on line 68?

Reply:
I copy-pasta'd this from `keyboard.ex`.  I don't know what @vanvoljg's original reason was to remove the use of `mod`, but it turned out to remove a dependency and a file.  (I delete `Xebow.Utils`, because nothing needed `mod` anymore; there is still an `RGBMatrix.Utils` that has `mod` in it that's used by the animations, but the entire `RGBMatrix` namespace is eventually going to be pulled out into a separate library, so using its `mod` here would be awkward.  I really wish Elixir had a `mod` built in that acted like all other languages...)

Reply:
The logic was to make it more readable. It does the same as what `mod` would have done, since there's a conditional in there, too, but @amclain thought this was more explicit, and I agreed. Plus, it means we don't need to use the additional utils.

Reply:
>The logic was to make it more readable [...] @amclain thought this was more explicit

Yeah, my reasoning is that it would be easier for someone to understand the value is rolling over if they see it set back to `0`, rather than having to decipher the result of `mod`.

>the entire `RGBMatrix` namespace is eventually going to be pulled out into a separate library, so using its mod here would be awkward

That turned out to be the other nice benefit. I think we legitimately need `mod` in the animation implementations due to animation math things (that's a technical term). It felt less necessary on this side of the app when handling the rollover, so it was nice to be able to drop the dependency on it.